src/libcore/iter/mod.rs

```diff
@@ -579,15 +579,15 @@ impl<'a, I, T: 'a> FusedIterator for Cloned<I>
 {}
 
 #[doc(hidden)]
-default unsafe impl<'a, I, T: 'a> TrustedRandomAccess for Cloned<I>
+unsafe impl<'a, I, T: 'a> TrustedRandomAccess for Cloned<I>
```

Comment:
Indeed, I think `default` here was incorrect. 

cc @rust-lang/libs -- `TrustedRandomAccess` was *meant* to be implemented for `Cloned<I>`, right?

(i.e., a default impl is not itself an impl, it just supplies defaults to other impls)

Reply:
Yeah I think this looks fine by me!